Add Payjoin project page (#637)
OpenSats/website: data/projects/payjoin.mdx

---
title: 'Payjoin'
dateAdded: '2023-07-14'
summary: 'A privacy-preserving batching protocol for Bitcoin transactions.'
nym: 'Dan Gould'
website: 'https://payjoin.org/'
donationLink: 'https://payjoin.org/donate'
coverImage: '/static/images/projects/payjoin.svg'
git: 'https://github.com/payjoin'
twitter: 'payjoindevkit'
tags: ['Bitcoin', 'Privacy']
showcase: true
fund: general
announcementLink: '/blog/bitcoin-grants-july-2023#payjoin-dev-kit'
---

[Payjoin](https://payjoin.org/) is a protocol that lets the sender and
receiver of a Bitcoin transaction cooperate to build a single transaction
with inputs from both parties. The resulting transaction breaks the
[common-input-ownership heuristic](https://en.bitcoin.it/wiki/Privacy#Common-input-ownership_heuristic)
that chain analysis tools rely on, and at the same time batches two payments
into one, which saves on fees. Used widely, Payjoin makes the common-input
heuristic useless across the whole network, not only for the people using it.

> The [Payjoin Foundation](https://payjoin.org/) is the 501(c)(3) non-profit

Reviewer: Same link as the first one?

stewarding this work, incorporated in Delaware in 2025 and led by
[Dan Gould](https://github.com/DanGould). Its main project is the
[Payjoin Dev Kit](https://github.com/payjoin): a Rust implementation
([rust-payjoin](https://github.com/payjoin/rust-payjoin)) with bindings for
Python, Dart, Swift, Kotlin, JavaScript, and C#. The dev kit powers
[BIP 77](https://github.com/bitcoin/bips/blob/master/bip-0077.md),
Async Payjoin, which uses
[Oblivious HTTP](https://www.rfc-editor.org/rfc/rfc9458.html) so sender and
receiver can exchange messages without both being online at once. That
removed the biggest practical barrier to adoption: Payjoin v1 needed the
receiver to run a public server, often behind Tor and TLS, which very few
wallets ever did.

## Why fund it?

Bitcoin privacy on-chain needs widespread adoption of tools that actively
break surveillance heuristics. A clean, production-ready library gives wallet
and service developers a drop-in way to add this behavior by default, without
each project building its own implementation.

> OpenSats first funded this work as the Payjoin Dev Kit in the

Reviewer: ```suggestion
OpenSats started funding this work as the Payjoin Dev Kit in the
```

[first wave of Bitcoin grants](/blog/bitcoin-grants-july-2023#payjoin-dev-kit)
in July 2023, and renewed the grant in the
[eleventh wave](/blog/eleventh-wave-of-bitcoin-grants) and again in the

> [seventeenth wave](/blog/seventeenth-wave-of-bitcoin-grants) now directly to

Reviewer: ```suggestion
[seventeenth wave](/blog/seventeenth-wave-of-bitcoin-grants), now directly awarded to
```

## What's next?

[Bull Bitcoin](https://bullbitcoin.com/) shipped Payjoin v2 in December 2024
and [Cake Wallet](https://cakewallet.com/) followed in May 2025.
[BDK-CLI](https://github.com/bitcoindevkit/bdk-cli) shipped v3.0.0 with
Payjoin support in early 2026. Pull requests are open for
[Liana](https://github.com/wizardsardine/liana/),
[LDK-Node](https://github.com/lightningdevkit/ldk-node),
[Boltz](https://boltz.exchange/), and [LEXE](https://lexe.app/), and a
[BTCPayServer](https://btcpayserver.org/) plugin is in development.

The `payjoin` crate has moved from version 0.11 to
[1.0-rc.2](https://github.com/payjoin/rust-payjoin/releases/tag/payjoin-1.0.0-rc.2),
adding persistence baked into the core state machine, a session-event log so
apps can show transaction history, and stable FFI for every binding target. [`payjoin-mailroom`](https://github.com/payjoin/rust-payjoin/tree/master/payjoin-mailroom)
bundles the directory, OHTTP relay, and metrics into a single binary that
[BOB Space](https://bobspaces.net/), Cake Wallet, Vinteum, Achow101, and
others now run as public infrastructure.

Alongside the dev kit, the Foundation runs a research track led by Armin

> Sabouri on wallet fingerprinting

Reviewer: ```suggestion
Alongside the dev kit, the Foundation runs a research track led by 
[Armin Sabouri](https://github.com/arminsabouri) on wallet fingerprinting
```
